gdal/frmts/grass/grass57dataset.cpp

```diff
@@ -454,23 +416,23 @@ CPLErr GRASSRasterBand::ResetReading ( struct Cell_head *sNewWindow )
          sNewWindow->ew_res != sOpenWindow.ew_res || sNewWindow->ns_res != sOpenWindow.ns_res ||
          sNewWindow->rows   != sOpenWindow.rows   || sNewWindow->cols   != sOpenWindow.cols )
     {
-        SetWindow ( sNewWindow );
-        G_copy((void *) &sOpenWindow, (void *) sNewWindow, sizeof(struct Cell_head));
+        Rast_set_window ( sNewWindow );
```

Comment:
this change is wrong. The call to the C++ SetWindow() function is intended.

Reply:
Thanks, hope I got it right in the updated version (now: grass.cpp)

Reply:
Now I got it thanks to the reference 28a38e3. Fixed and pytest passes again.